No tag deleted when the delete is triggered by replication

Steps to reproduce.
1. Create a push based replication rule from HarborA/library/** to HarborB/pushed/**, and set the replicate based on event and allow deletion. the registry user is admin.
2. Push image HarborA/library/zdj:1.0
3. Login to HarborB, check the logs, there is a create log in HarborB.
4. Login to HarborB, delete the image HarborA/library/zdj:1.0 and delete the repo HarborA/library/zdj
5. Login to HarborB, the pushed/zdj:1.0 is not deleted.
6. Check the replication log in HarborA. 

```
2020-04-10T08:01:15Z [INFO] [/replication/transfer/image/transfer.go:115]: client for source registry [type: harbor, URL: http://core:8080, insecure: true] created
2020-04-10T08:01:15Z [INFO] [/replication/transfer/image/transfer.go:125]: client for destination registry [type: harbor, URL: https://10.193.23.3, insecure: true] created
2020-04-10T08:01:15Z [ERROR] [/replication/transfer/image/transfer.go:346]: failed to delete the manifest of artifact pushed/zdj:1.0 on the destination registry: <nil>, UNAUTHORIZED, http status code: 401, body: {"errors":[{"code":"UNAUTHORIZED","message":"unauthorized to access repository: pushed/zdj, action: delete"}]}
```

Expected Result:
The delete operation should be triggered by replication and success.

--- a/src/common/security/v2token/context.go
+++ b/src/common/security/v2token/context.go
@@ -94,13 +94,16 @@ func New(ctx context.Context, name string, access []*registry_token.ResourceActi
 		}
 		actionMap := make(map[types.Action]struct{})
 		for _, a := range ac.Actions {
-			if a == "pull" || a == "*" {
+			switch a {
+			case "pull":
 				actionMap[rbac.ActionPull] = struct{}{}
-			}
-			if a == "push" || a == "*" {
+			case "push":
 				actionMap[rbac.ActionPush] = struct{}{}
-			}
-			if a == "scanner-pull" {
+			case "*":
+				actionMap[rbac.ActionPull] = struct{}{}
+				actionMap[rbac.ActionPush] = struct{}{}
+				actionMap[rbac.ActionDelete] = struct{}{}
+			case "scanner-pull":
 				actionMap[rbac.ActionScannerPull] = struct{}{}
 			}
 		}

--- a/src/common/security/v2token/context_test.go
+++ b/src/common/security/v2token/context_test.go
@@ -71,6 +71,11 @@ func TestAll(t *testing.T) {
 		},
 		{
 			resource: rbac.NewProjectNamespace(2).Resource(rbac.ResourceRepository),
+			action:   rbac.ActionDelete,
+			expect:   true,
+		},
+		{
+			resource: rbac.NewProjectNamespace(2).Resource(rbac.ResourceRepository),
 			action:   rbac.ActionScannerPull,
 			expect:   false,
 		},

